### PR TITLE
Pools tests

### PR DESCRIPTION
**Pull request by joeyyy09**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

This PR adds additional test cases and edge cases for the `pools` package, enhancing the  reliability of the pool implementation. 

Added test cases for the following functions/tests:

    TestPool
    TestPool_Reuse
    TestPool_ConcurrentAccess
    TestPool_LargeNumbers
    TestPool_NegativeNumbers
    TestPool_ZeroValue

#### Which issue(s) this PR fixes:
Fixes #

#### Special notes for your reviewer:

Please review the added test cases to ensure they cover all potential edge cases effectively.

#### Does this PR introduce a user-facing change?
NONE

### CODE REVIEW

File: pkg/utils/pools/pools_test.go [hunk 1]
```diff
@@ -53,3 +53,108 @@ func TestPool(t *testing.T) {
...
+
+	if pool.Get() != 0 {
+		t.Errorf("expected 0, got %d", pool.Get())
+	}
+
+	pool.Put(0)
+	if pool.Get() != 0 {
+		t.Errorf("expected 0, got %d", pool.Get())
+	}
+}
```
Comment (wzshiming) [inline pkg/utils/pools/pools_test.go:160]:
```suggestion
```

File: pkg/utils/pools/pools_test.go [hunk 2]
```diff
@@ -53,3 +53,108 @@ func TestPool(t *testing.T) {
...
+	// Check the reuse of the pool
+	seen[pool.Get()] = true
+	seen[pool.Get()] = true
+	seen[pool.Get()] = true
+
+	// Ensure we have seen all values
+	if !seen[1] || !seen[2] || !seen[3] {
+		t.Errorf("expected to see 1, 2, and 3, but got seen map: %v", seen)
+	}
+}
```
Comment (wzshiming) [inline pkg/utils/pools/pools_test.go:86]: It is repeated with TestPool

File: pkg/utils/pools/pools_test.go [hunk 3]
```diff
@@ -53,3 +53,108 @@ func TestPool(t *testing.T) {
...
+
+func TestPool_ConcurrentAccess(t *testing.T) {
+	pool := NewPool(func() int {
+		return 0
+	})
```
Comment (wzshiming) [inline pkg/utils/pools/pools_test.go:91]:
```suggestion
	var index int32
	pool := NewPool(func() int {
		return int(atomic.AddInt32(&index, 1))
	})
```

and check that the results are not duplicated

File: pkg/utils/pools/pools_test.go [hunk 4]
```diff
@@ -53,3 +54,35 @@ func TestPool(t *testing.T) {
...
+	for i := 0; i < 10; i++ {
+		val := pool.Get()
+		if val < 0 || val >= 10 {
+			t.Errorf("unexpected value %d", val)
+		}
+		if values[val] {
+			t.Errorf("duplicate value found: %d", val)
+		}
+		values[val] = true
+	}
```
Comment (wzshiming) [inline pkg/utils/pools/pools_test.go:87]: Check that the length of `values` is as expected
